CommonCompilerConfig.cmake

```diff
@@ -159,16 +183,24 @@ endif()
 # try compiling without them.
 #
 if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
-  set(optimization_flags_cc "${optimization_flags_cc} -mtune=generic -O2")
+  set(optimization_flags_cc "${optimization_flags_cc} -O2")
   set(optimization_flags_cc "-pipe ${optimization_flags_cc}") #TODO use -Ofast instead of -O3
   set(optimization_flags_lt "-O2 ${optimization_flags_lt}")
 
+  CHECK_CXX_COMPILER_FLAG(-mtune=generic COMPILER_SUPPORTS_GENERIC_TUNING)
```

Comment:
maybe add a #comment that some compilers (on ARM) do not support -mtune=generic? ..Or why do you need this safety check